waffle-mock-contract/test/directTest.ts

```diff
@@ -0,0 +1,132 @@
+      const counterFactory = new ContractFactory(Counter.abi, Counter.bytecode, wallet);
+      const counter = await counterFactory.deploy();
+      const mockCounter = await deployMockContract(wallet, Counter.abi, address);
+
+      expect(await mockCounter.staticcall(counter, 'read()')).to.equal('0');
+      expect(await mockCounter.staticcall(counter, 'read')).to.equal('0');
+    });
+
+    it('should be able to call another contract with a parameter', async () => {
+      const address = Wallet.createRandom().address;
+      const counterFactory = new ContractFactory(Counter.abi, Counter.bytecode, wallet);
+      const counter = await counterFactory.deploy();
+      const mockCounter = await deployMockContract(wallet, Counter.abi, address);
+
+      expect(await mockCounter.staticcall(counter, 'add', 1)).to.equal('1');
+    });
+
+    it('should be able to call another contract with many parameters', async () => {
+      const address = Wallet.createRandom().address;
+      const counterFactory = new ContractFactory(Counter.abi, Counter.bytecode, wallet);
+      const counter = await counterFactory.deploy();
+      const mockCounter = await deployMockContract(wallet, Counter.abi, address);
+
+      expect(await mockCounter.staticcall(counter, 'addThree', 1, 2, 3)).to.equal('6');
+    });
+
+    it('should be able to execute another contract', async () => {
+      const address = Wallet.createRandom().address;
+      const counterFactory = new ContractFactory(Counter.abi, Counter.bytecode, wallet);
+      const counter = await counterFactory.deploy();
+      const mockCounter = await deployMockContract(wallet, Counter.abi, address);
+
+      await mockCounter.call(counter, 'increment()');
+      expect(await counter.read()).to.equal('1');
+
+      await mockCounter.call(counter, 'increment');
+      expect(await counter.read()).to.equal('2');
+    });
+
+    it('should be able to execute another contract with a parameter', async () => {
+      const address = Wallet.createRandom().address;
+      const counterFactory = new ContractFactory(Counter.abi, Counter.bytecode, wallet);
+      const counter = await counterFactory.deploy();
+      const mockCounter = await deployMockContract(wallet, Counter.abi, address);
+
+      await mockCounter.call(counter, 'increaseBy', 2);
+      expect(await counter.read()).to.equal('2');
+    });
+
+    it('should be able to execute another contract with many parameters', async () => {
+      const address = Wallet.createRandom().address;
+      const counterFactory = new ContractFactory(Counter.abi, Counter.bytecode, wallet);
+      const counter = await counterFactory.deploy();
+      const mockCounter = await deployMockContract(wallet, Counter.abi, address);
+
+      await mockCounter.call(counter, 'increaseByThreeValues', 1, 2, 3);
+      expect(await counter.read()).to.equal('6');
+    });
+
+    it('can be deployed under specified address', async () => {
```

Comment:
I think it's a good idea to also have the intended flow, meaning that we have hardcoded address in some contract and we want to mock the contract under this hardcoded address. Then the first contract calls the mocked contract and gets some result, using it further in it's computations (can be as simple as `require(result == 1)`)